### PR TITLE
removed sync, added optional signing and multiple fixes to CI pipeline

### PR DESCRIPTION
**Pull request by mihaiplesa**
Includes workaround for https://github.com/brave/brave-browser/issues/1922
Closes https://github.com/brave/brave-browser/issues/4665
Closes https://github.com/brave/brave-browser/issues/4666
Closes https://github.com/brave/brave-browser/issues/4667
Closes https://github.com/brave/brave-browser/issues/4668

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Windows
  - [x] macOS
  - [x] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [x] Windows
  - [x] macOS
  - [x] Linux
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions.

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by petemill** [comment]
Please can we keep in `sync -- --all` instead of always running `init`? Was there an issue this is attempting to address?

This would remove benefits of upcoming when the rewrite of https://github.com/brave/brave-browser/pull/3185 lands. This PR aims to _only_ update the chromium files whose brave patches have changed. That, coupled with running `sync -- --all` and not `init` should greatly reduce incremental build time even when switching branches.

**Comment by mihaiplesa** [comment]
Thanks for the insight @petemill it's paused for now but we need a solution for the version mismatches we keep on getting. the safest is to wipe the workspace at every build as not even running init again fixes it. Or maybe just removing `src/brave` and doing init would do.

**Comment by petemill** [comment]
@mihaiplesa thanks for the info. Removing `src/brave` would certainly also exhibit the problem of incremental builds not working!

Is there an issue describing the problem we're seeing with version mismatches. As far as I know that's mostly caused by the brave-core branch not being re-based on master - something that has to happen manually, unless there's a pinned brave-browser branch. Is there something else?

**Comment by mihaiplesa** [comment]
@petemill will check if there's an issue created. Even if the branches are rebased and versions match and doing init and sync (with or without using git cache) the mismatch doesn't go away.

**Comment by mihaiplesa** [comment]
@petemill issue is https://github.com/brave/brave-browser/issues/4520

File: Jenkinsfile [hunk 1]
```diff
@@ -7,10 +7,11 @@ pipeline {
     }
     parameters {
         choice(name: "CHANNEL", choices: ["nightly", "dev", "beta", "release"], description: "")
+        choice(name: "BUILD_TYPE", choices: ["Release", "Debug"], description: "")
```
Comment (bsclifton) [inline Jenkinsfile]: How does default value get set? (asked same question in https://github.com/brave/brave-core/pull/2443/files#r289559506). If I check in `brave-browser`, how does it know which build type to default to?

Comment (mihaiplesa) [reply Jenkinsfile]: first in the list is default

File: Jenkinsfile [hunk 2]
```diff
@@ -417,22 +439,21 @@ pipeline {
...
-                                sh "npm run init"
-                            }
-                        }
-                        stage("sync") {
-                            steps {
-                                sh "npm run sync -- --all"
+                                sh """
+                                    set -e
+                                    rm -rf src/brave
+                                    npm run init
```
Comment (bsclifton) [inline Jenkinsfile]: 😎

**Pull request merged**